app/src/main/java/com/github/mik629/android/fundamentals/di/movies_list/MoviesListViewModelComponent.kt

```diff
@@ -0,0 +1,12 @@
+package com.github.mik629.android.fundamentals.di.movies_list
+
+import com.bsquaredwifi.app.di.scopes.VM
+import com.github.mik629.android.fundamentals.di.AppComponent
+import com.github.mik629.android.fundamentals.ui.movieslist.MoviesListViewModel
+import dagger.Component
+
+@VM
+@Component(dependencies = [AppComponent::class])
+interface MoviesListViewModelComponent {
+    val viewModel: MoviesListViewModel
```

Comment:
in this case better to write fun instead of val

Reply:
why?

Reply:
when u read SomeComponent.viewModel
u think about viewModel as property, but not as some fun that can create or get from cache viewModel
just idiomatic